service/frontend/workflowHandler_test.go

```diff
@@ -636,6 +636,93 @@ func (s *workflowHandlerSuite) TestRecordActivityTaskHeartbeat_TaskTokenNotSet()
 	s.Nil(result)
 }
 
+func (s *workflowHandlerSuite) TestRecordActivityTaskHeartbeatByID_Success() {
+	wh := s.getWorkflowHandler(s.newConfig(dc.NewInMemoryClient()))
+	req := &types.RecordActivityTaskHeartbeatByIDRequest{
+		Domain:     s.testDomain,
+		WorkflowID: testWorkflowID,
+		RunID:      testRunID,
+		ActivityID: "1",
+		Details:    nil,
+		Identity:   "",
```

Comment:
I suggest to only initialise fields with non-default values unless we're already doing the same in other tests of workflowHandler.
It is expected to have Details and Identity not initialised, and you don't usually want to emphasise on this.